packages/preview/linguify/0.4.0/lib/linguify.typ

```diff
@@ -0,0 +1,163 @@
+// linguify.typ
+#import "utils.typ": *
+#import "fluent.typ"
+
+/// None or dictionary of the following structure:
+///
+/// - `conf`
+///   - `data_type` (string): The type of data structure used for the database. If not specified, it defaults to `dict` structure.
+///   - `default-lang` (string): The default language to use as a fallback if the key in the preferred language is not found.
+///   - ...
+/// - `lang`
+///   - `en`: The English language section.
+///   - ...
+#let database = state("linguify-database", none)
+
+
+/// Set the default linguify database
+///
+/// The data must contain at least a lang section like described at @@database.
+///
+/// - data (dictionary): the database which will be set to @@database
+/// -> content (state-update)
+#let set_database(data) = {
+  assert.eq(type(data), dictionary, message: "expected data to be a dictionary, found " + type(data))
+  if (data.at("conf", default: none) == none) {
+    data.insert("conf", (:))
+  }
+  // assert(data.at("conf", default: none) != none, message: "missing conf section ")
+  assert(data.at("lang", default: none) != none, message: "missing lang section ")
+
+  database.update(data);
+}
+
+/// Add data to the current database
+///
+/// - data (dictionary): the database which will be added to @@database
+/// -> content (state-update)
+#let update_database(data) = {
+  context {
+    let _database = database.get()
+    for (key,value) in data.pairs() {
+      // let lang_section = database.at(key, default: none)
+      if key not in _database.keys() {
+        _database.insert(key, value)
+      } else {
+        let new = _database.at(key) + value
+        _database.insert(key, new)
+      }
+    }
+    database.update(_database);
+  }
```

Comment:
Didn't see this before, but I recommend using an update closure instead of get() + update(). Calling `update_database` more than five times will likely lead to a convergence error.

Reply:
Ok thanks for the info, i will change this later. ;)

Reply:
Decided to remove this function, as it does not fit for all current usecases anymore.